docs/tutorials/agentic_memory_with_strands_agent.md

```diff
@@ -0,0 +1,310 @@
+      "embeddingTypes": ["float"]
+    },
+    "credential": {
+      "access_key": "your_aws_access_key",
+      "secret_key": "your_aws_secret_key",
+      "session_token": "your_aws_session_token"
+    },
+    "actions": [{
+      "action_type": "predict",
+      "method": "POST",
+      "url": "https://bedrock-runtime.${parameters.region}.amazonaws.com/model/${parameters.model}/invoke",
+      "headers": {"content-type": "application/json", "x-amz-content-sha256": "required"},
+      "request_body": "{ \"inputText\": \"${parameters.inputText}\", \"dimensions\": ${parameters.dimensions}, \"normalize\": ${parameters.normalize}, \"embeddingTypes\": ${parameters.embeddingTypes} }",
+      "pre_process_function": "connector.pre_process.bedrock.embedding",
+      "post_process_function": "connector.post_process.bedrock.embedding"
+    }]
+  }
+}
+```
+
+## 3. Create LLM for Fact Extraction
+
+Register a Bedrock Claude model for fact extraction:
+
+```
+POST /_plugins/_ml/models/_register
+{
+  "name": "Bedrock infer model",
+  "function_name": "remote",
+  "description": "LLM model for memory processing",
+  "connector": {
+    "name": "Amazon Bedrock Connector: LLM",
+    "description": "The connector to bedrock Claude 3.7 sonnet model",
+    "version": 1,
+    "protocol": "aws_sigv4",
+    "parameters": {
+      "region": "us-east-1",
+      "service_name": "bedrock",
+      "max_tokens": 8000,
+      "temperature": 1,
+      "anthropic_version": "bedrock-2023-05-31",
+      "model": "us.anthropic.claude-3-7-sonnet-20250219-v1:0"
+    },
+    "credential": {
+      "access_key": "your_aws_access_key",
+      "secret_key": "your_aws_secret_key",
+      "session_token": "your_aws_session_token"
+    },
+    "actions": [{
+      "action_type": "predict",
+      "method": "POST",
+      "headers": {"content-type": "application/json"},
+      "url": "https://bedrock-runtime.${parameters.region}.amazonaws.com/model/${parameters.model}/invoke",
+      "request_body": "{ \"system\": \"${parameters.system_prompt}\", \"anthropic_version\": \"${parameters.anthropic_version}\", \"max_tokens\": ${parameters.max_tokens}, \"temperature\": ${parameters.temperature}, \"messages\": ${parameters.messages} }"
+    }]
+  }
+}
+```
+
+## 4. Create Memory Container
+
+Create a memory container, using the two models we created in the previous steps:
+
+```
+POST /_plugins/_ml/memory_containers/_create
+{
+  "name": "memory container",
+  "description": "Memory container",
+  "memory_storage_config": {
+    "llm_model_id": "your_llm_model_id",
+    "embedding_model_type": "TEXT_EMBEDDING",
+    "embedding_model_id": "your_embedding_model_id",
+    "dimension": 1024
+  }
+}
+```
+
+Sample response:
+```
+{ 'memory_container_id': 'xthvpZgBOh0h20Y9O3Sa', 'status': 'created' }
+```
+
+## 5. Implement Strands Agent with Memory Tools
+
+We will now create a memory agent using the [Strands Agents](https://strandsagents.com/latest/) framework and OpenSearch agentic memory.
```

Comment:
Can you explain this is generic and can be used for any agent framework? Here we just use Strands Agent to demonstrate